sdetools/alm_integration/tests/alm_integration_plugin_test_base.py

```
@@ -0,0 +1,114 @@
+import unittest
+
+from sdetools.alm_integration.alm_integration_plugin_base import BaseALMIntegration, IntegrationField, FieldType, Option, Section
+
+from sdetools.sde_plugins.exceptions import PluginException
+
+
+class ALMIntegrationPluginBaseTests(unittest.TestCase):
+
+    # note: we should stay consistent when returning field.name or field.label in our error msgs
+    REQUIRED_ERROR = 'Field Test Field is required.'
+    MAX_LENGTH_ERROR = 'Max length of field Test Field is 5.'
+    OPTION_ERROR = 'almyo is an invalid option for field conflict_policy'
+    DISABLED_ERROR = 'The following field is disabled: readonly_field'
+    BASE_VALIDATE_ERRORS = [
+        'Field ALM Project is required.',
+        'Max length of field Test Field is 5.'
+    ]
+
+    def setUp(self):
+        self.mock = MockBaseALMIntegration()
+
+    def test_base_exclude_required_validation(self):
```

I think it is better to let the test fail if an error is raised and assume a pass otherwise, it makes the code much more readable. Although, I am open to suggestions :)

Discussion here:
http://stackoverflow.com/questions/4319825/python-unittest-opposite-of-assertraises